deploy/charts/istio-csr/templates/deployment.yaml

```diff
@@ -120,6 +120,18 @@ spec:
         resources:
           {{- toYaml .Values.resources | nindent 12 }}
 
+        securityContext:
+          allowPrivilegeEscalation: false
+          readOnlyRootFilesystem: true
+          runAsNonRoot: true
+          capabilities:
+            drop:
+            - ALL
```

Comment:
suggestion: I think it's a good idea to enable the security context and set sane defaults - clearly these are pretty good defaults since the e2e tests passed without an issue.

That said, I think setting a restrictive securityContext like this - with no option to disable it - might cause issues for some users.

Can we add the securityContext to the values.yaml? I think it seems fine to have these defaults in values.yaml, but at least that will allow users to opt-out!

Reply:
Yep, kinda autopiloted copying Istio's helm charts but this makes more sense